third-party/cmake/DownloadAndExtractFile.cmake

```diff
@@ -85,6 +75,25 @@ if(NOT status_code EQUAL 0)
 ")
 endif()
 
+set(NULL_SHA1 "0000000000000000000000000000000000000000")
+
+# We could avoid computing the SHA1 entirely if a NULL_SHA1 was given,
+# but we want to warn users of an empty file.
+file(SHA1 ${file} ACTUAL_SHA1)
+if(ACTUAL_SHA1 STREQUAL "da39a3ee5e6b4b0d3255bfef95601890afd80709")
+  # File was empty.  It's likely due to lack of SSL support.
+  message(FATAL_ERROR
+    "Failed to download ${URL}.  The file is empty and likely means CMake "
+    "was built without SSL support.  Please use a version of CMake with "
+    "proper SSL support.")
+elseif((NOT EXPECTED_SHA1 STREQUAL NULL_SHA1) AND
+       (NOT EXPECTED_SHA1 STREQUAL ACTUAL_SHA1))
+  # Wasn't a NULL SHA1 and we didn't match, so we fail.
```

Comment:
The user facing message is for something else (the sha1 of an empty file).  This is about the all zeros sha1, which we use as a wildcard.